src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.cc

```diff
@@ -229,15 +266,49 @@ XdsOverrideHostLb::Picker::Picker(
   }
 }
 
+/*
+    // No READY subchannel found.  If we found an IDLE subchannel,
+    // trigger a connection attempt and queue the pick until that attempt
+    // completes.
+    if idle_subchannel is not None:
+      hop into control plane to trigger connection attempt for idle_subchannel
+      return queue as pick result
+*/
+
+absl::optional<LoadBalancingPolicy::PickResult>
+XdsOverrideHostLb::PickOverridenHost(absl::string_view address) {
+  if (address.length() == 0) {
+    return absl::nullopt;
+  }
+  MutexLock lock(&subchannel_map_mu_);
+  auto it = subchannel_map_.find(address);
+  if (it == subchannel_map_.end()) {
+    return absl::nullopt;
+  }
+  auto subchannel = it->second.GetSubchannel();
+  if (subchannel == nullptr) {
+    return absl::nullopt;
+  }
+  auto connectivity_state = subchannel->connectivity_state();
```

Comment:
This requires synchronization.  The connectivity state notifications that set this value will run in the control plane `WorkSerializer`, but this code that reads them will be run in the picker, which does not run in the `WorkSerializer`, so there can be a read and a write in different threads at the same moment.

The easiest thing would probably be to change the type of `SubchannelWrapper::connectivity_state_` to `std::atomic<grpc_connectivity_state>`.